### PR TITLE
[GLUTEN-1207] Fix a dependency issue for spark shims

### LINKED ISSUE #1207: shims-spark32 and shims-spark33 may be depencied at the same time
**Describe the bug**

I  build for spark-3.3, and run tpcds tests, it reports following error.

```
scala> time{spark.sql(fileContents).show}
java.lang.ClassCastException: scala.Tuple8 cannot be cast to scala.Tuple7
  at io.glutenproject.extension.JoinSelectionOverrides.$anonfun$apply$1(StrategyOverrides.scala:186)
  at io.glutenproject.utils.QueryPlanSelector.maybeNil(QueryPlanSelector.scala:73)
  at io.glutenproject.extension.JoinSelectionOverrides.apply(StrategyOverrides.scala:180)
  at org.apache.spark.sql.catalyst.planning.QueryPlanner.$anonfun$plan$1(QueryPlanner.scala:63)
  at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:486)
  at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:492)
  at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
  at org.apache.spark.sql.catalyst.planning.QueryPlanner.plan(QueryPlanner.scala:93)
  at org.apache.spark.sql.execution.SparkStrategies.plan(SparkStrategies.scala:69)
  at org.apache.spark.sql.catalyst.planning.QueryPlanner.$anonfun$plan$3(QueryPlanner.scala:78)
  at scala.collection.TraversableOnce$folder$1.apply(TraversableOnce.scala:196)
  at scala.collection.TraversableOnce$folder$1.apply(TraversableOnce.scala:194)
  at scala.collection.Iterator.foreach(Iterator.scala:943)
  at scala.collection.Iterator.foreach$(Iterator.scala:943)
  at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
  at scala.collection.TraversableOnce.foldLeft(TraversableOnce.scala:199)
  at scala.collection.TraversableOnce.foldLeft$(TraversableOnce.scala:192)
  at scala.collection.AbstractIterator.foldLeft(Iterator.scala:1431)
  at org.apache.spark.sql.catalyst.planning.QueryPlanner.$anonfun$plan$2(QueryPlanner.scala:75)
  at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:486)
  at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:492)
  at org.apache.spark.sql.catalyst.planning.QueryPlanner.plan(QueryPlanner.scala:93)
```

**To Reproduce**
```
mvn clean install -Pbackends-velox -Pspark-3.2 -DskipTests
mvn clean install -Pbackends-velox -Pspark-3.3 -DskipTests
```

I will make a pr fix it.

### PR DESCRIPTION
**Pull request by sharkdtu**
## What changes were proposed in this pull request?

FIx #1207

## How was this patch tested?

manual tests

### CODE REVIEW

**Comment by PHILO-HE** [comment]
Thanks for your PR.
I note there is an issue previously, looks similar to yours: https://github.com/oap-project/gluten/issues/650.
Do your actually deploy the expected jar for your spark? I.e., build gluten with spark-3.3 option and deploy the jar for spark 3.3. Thanks!

**Comment by sharkdtu** [comment]
@PHILO-HE 
I have build for spark3.3, and run tcpds tests with `spark-3.3.2-bin-hadoop2`.
I found shims-spark32 was dependencied by default, and can not be overwritten by profile `-Pspark3.3` . It may be conflict when packaging.

**Comment by PHILO-HE** [comment]
Thanks for the feedback.
The patch looks good to me. But I am still wondering why spark 3.2 shim is introduced in building for spark 3.3. By checking the jar for spark 3.3 (`jar -tf GLUTEN_JAR | grep shim`), it looks spark 3.2 shim is not included. Am I missing something?

File: pom.xml [hunk 1]
```diff
@@ -103,6 +103,7 @@
     <maven.jar.plugin>3.2.2</maven.jar.plugin>
     <scalastyle.version>1.0.0</scalastyle.version>
     <scalatest-maven-plugin.version>2.2.0</scalatest-maven-plugin.version>
+    <shims-spark-artifactId>spark-sql-columnar-shims-spark32</shims-spark-artifactId>
```
Comment (Yohahaha) [inline pom.xml:106]: Why this line needed?

**Comment by sharkdtu** [comment]
@PHILO-HE 
Maybe it is maven version problem, I used apache-maven-3.8.5, and print dependencies by `mvn -Pbackends-velox -Pspark-3.3 dependency:tree`. 
```
[INFO] Building Gluten Core 0.5.0-SNAPSHOT                                [5/9]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ gluten-core ---
[INFO] io.glutenproject:gluten-core:jar:0.5.0-SNAPSHOT
[INFO] +- org.apache.celeborn:celeborn-client-spark-3-shaded_2.12:jar:0.2.1-SNAPSHOT:compile
[INFO] +- io.glutenproject:spark-sql-columnar-shims-spark33:jar:0.5.0-SNAPSHOT:compile
...

[INFO] Building Gluten Data 0.5.0-SNAPSHOT                                [6/9]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ gluten-data ---
[INFO] io.glutenproject:gluten-data:jar:0.5.0-SNAPSHOT
[INFO] +- org.apache.celeborn:celeborn-client-spark-3-shaded_2.12:jar:0.2.1-SNAPSHOT:compile
[INFO] +- io.glutenproject:gluten-core:jar:0.5.0-SNAPSHOT:compile
[INFO] |  +- io.glutenproject:spark-sql-columnar-shims-common:jar:0.5.0-SNAPSHOT:compile
[INFO] |  +- com.google.protobuf:protobuf-java:jar:3.16.3:compile
[INFO] |  +- com.google.protobuf:protobuf-java-util:jar:3.16.3:compile
[INFO] |  |  +- com.google.guava:guava:jar:31.1-jre:provided
[INFO] |  |  |  +- com.google.guava:failureaccess:jar:1.0.1:provided
[INFO] |  |  |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:provided
[INFO] |  |  |  +- org.checkerframework:checker-qual:jar:3.12.0:provided
[INFO] |  |  |  \- com.google.j2objc:j2objc-annotations:jar:1.3:provided
[INFO] |  |  +- com.google.errorprone:error_prone_annotations:jar:2.3.4:compile
[INFO] |  |  \- com.google.code.gson:gson:jar:2.8.6:compile
[INFO] |  \- io.glutenproject:spark-sql-columnar-shims-spark32:jar:0.5.0-SNAPSHOT:compile
```

**Review by PHILO-HE — APPROVED** [review]

**Comment by PHILO-HE** [comment]
Will merge the patch soon. Thanks for the contribution!

**Pull request merged**